generators/php/model/src/object/ObjectGenerator.ts

```diff
@@ -0,0 +1,179 @@
+                    arguments: [`DateType::TYPE_${underlyingInternalType.type.toUpperCase()}`]
+                })
+            );
+        }
+        if (underlyingInternalType.type === "array" || underlyingInternalType.type === "map") {
+            attributes.push(
+                php.attribute({
+                    reference: this.context.getArrayTypeClassReference(),
+                    arguments: [this.getArrayTypeAttributeArgument(type.underlyingType())]
+                })
+            );
+        }
+        return attributes;
+    }
+
+    private getArrayTypeAttributeArgument(type: php.Type): php.AstNode {
+        switch (type.internalType.type) {
+            case "int":
+                return php.codeblock('"integer"');
+            case "string":
+                return php.codeblock('"string"');
+            case "bool":
+                return php.codeblock('"bool"');
+            case "float":
+                return php.codeblock('"float"');
+            case "date":
+                return php.codeblock('"date"');
+            case "dateTime":
+                return php.codeblock('"datetime"');
+            case "mixed":
+                return php.codeblock('"mixed"');
+            case "object":
+                // this is likely not handled by our serde, but we also never use it
+                return php.codeblock('"object"');
+            case "array":
+                return php.array({
+                    entries: [this.getArrayTypeAttributeArgument(type.internalType.value)]
+                });
+            case "map": {
+                return php.map({
+                    entries: [
+                        {
+                            key: this.getArrayTypeAttributeArgument(type.internalType.keyType),
+                            value: this.getArrayTypeAttributeArgument(type.internalType.valueType)
+                        }
+                    ]
+                });
+            }
+            case "typeDict": {
+                return php.map({
+                    entries: [
+                        {
+                            key: php.codeblock('"string"'),
+                            value: php.codeblock('"mixed"')
+                        }
+                    ]
+                });
+            }
+            case "optional":
+                return php.instantiateClass({
+                    classReference: this.context.getUnionClassReference(),
+                    arguments_: [this.getArrayTypeAttributeArgument(type.internalType.value), php.codeblock('"null"')]
+                });
+            case "reference": {
+                const reference = type.internalType.value;
+                return php.codeblock((writer) => {
+                    writer.writeNode(reference);
+                    writer.write("::class");
+                });
+            }
+            default:
+                assertNever(type.internalType);
+        }
+    }
+}
```

Comment:
Discussed offline - 

These functions might better belong alongside the Type definitions, but the configurable core namespace makes this a little tricky. Given that these attributes always pertain to object generation, we can leave this as-is for now but it might be worth revisiting.